### PR TITLE
update numpy

### LINKED ISSUE #2105: Less strict numpy version pinning
## 🚀 Feature Request
Allow for newer numpy versions than `<1.23.0`.

## Motivation

Currently, composer fixes numpy to be `'numpy>=1.21.5,<1.23.0'`. This is unfortunate, because other requirements that we use need numpy > 1.23.0, creating an incompatibility.

This was set in #1345 in reaction to a build failure because scipy need(ed) `<1.23.0`. Scipy itself is apparently pulled from `torch-metrics`. 

I don't believe this is current anymore:
- composer pulls in `torchmetrics>=0.10.0,<0.11.4`
- torchmetrics 0.11.3 pulls in `scipy >1.0.0, <=1.10.0` [(requirements)](https://github.com/Lightning-AI/metrics/blob/v0.11.3/requirements/image.txt#LL4C23-L4C23)
- scipy 1.10.0 sets `np_maxversion = '1.27.0'` [(setup.py)](https://github.com/scipy/scipy/blob/v1.10.0/setup.py#L453)

Indeed, scipy has https://github.com/scipy/scipy/issues/16964, which was fixed in https://github.com/scipy/scipy/pull/16966 in September of 2022, while #1345 above was noted in August.

Accordingly, it looks like relaxing the numpy maxversion should have a valid requirements path. 

The closed PR https://github.com/mosaicml/composer/pull/1835 would already implement this.

### PR DESCRIPTION
**Pull request by dakinggg**
# What does this PR do?
Updates the numpy version requirement to be up to date. Waiting to see if CI passes before requesting review.

# What issue(s) does this change relate to?
Closes [CO-1974](https://mosaicml.atlassian.net/browse/CO-1974)
Closes #2105 

# Before submitting
- [x] Have you read the [contributor guidelines](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md)?
- [x] Was this change discussed/approved in a GitHub issue first? It is much more likely to be merged if so.
- [x] Did you update any related docs and document your change?
- [x] Did you update any related tests and add any new tests related to your change? (see [testing](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md#running-tests))
- [ ] Did you run the tests locally to make sure they pass?
- [x] Did you run `pre-commit` on your change? (see the `pre-commit` section of [prerequisites](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md#prerequisites))

[CO-1974]: https://mosaicml.atlassian.net/browse/CO-1974?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by dakinggg** [comment]
bump @mvpatel2000

**Review by mvpatel2000 — APPROVED** [review]
Why are type ignores needed?

**Comment by dakinggg** [comment]
It was a hard to parse error related to numpy types
